cmd/controller/main.go

```diff
@@ -15,15 +15,25 @@ limitations under the License.
 package main
 
 import (
-	"context"
+	"fmt"
 
 	"github.com/aws/karpenter/pkg/cloudprovider"
 	"github.com/aws/karpenter/pkg/cloudprovider/aws"
 	"github.com/aws/karpenter/pkg/controllers"
+	"github.com/aws/karpenter/pkg/startup"
 )
 
 func main() {
-	controllers.Initialize(func(ctx context.Context, options cloudprovider.Options) cloudprovider.CloudProvider {
-		return aws.NewCloudProvider(ctx, options)
-	})
+	options := startup.Initialize()
+	cloudProvider := startup.Decorate(aws.NewCloudProvider(options.Ctx, cloudprovider.Options{
+		ClientSet:  options.Clientset,
```

Comment:
If cloudProvider.Options is taking so many fields off of this, does it make sense to do something like

type cloudprovider.Options struct {
  controller.Options
}